src/NUnitFramework/framework/Constraints/SubstringConstraint.cs

```diff
@@ -43,22 +44,34 @@ public SubstringConstraint(string expected) : base(expected)
         }
 
         /// <summary>
+        /// Modify the constraint to ignore case in matching.
+        /// </summary>
+        public override StringConstraint IgnoreCase
+        {
+            get { caseInsensitive = true; comparisonType = null; return this; }
```

Comment:
@CharliePoole Is this what we want? As we discussed earlier and I double checked via testing, currently `Is.EqualTo(x).Using(StringComparer.Ordinal).IgnoreCase` does *not* reset the `Using` setting.. all `IgnoreCase` calls are ignored if any `Using` is present no matter the order.

I'd be more in favor of throwing an exception, if anything.